checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -201,6 +209,108 @@ private void checkOwningOverrides(
+   * @param mcAtf the MustCallAnnotatedTypeFactory
+   */
+  private void checkMustCallAliasAnnotationForMethod(
+      MethodTree tree, MustCallAnnotatedTypeFactory mcAtf) {
+
+    Element paramWithMustCallAliasAnno = getParameterWithMustCallAliasAnno(tree);
+    boolean isMustCallAliasAnnoOnParameter = paramWithMustCallAliasAnno != null;
+
+    if (TreeUtils.isVoidReturn(tree) && isMustCallAliasAnnoOnParameter) {
+      checker.reportWarning(
+          tree, "mustcallalias.method.return.and.param", "this method has a void return");
+      return;
+    }
+
+    AnnotatedTypeMirror returnType = mcAtf.getMethodReturnType(tree);
+    boolean isMustCallAliasAnnoOnReturnType = returnType.hasPrimaryAnnotation(PolyMustCall.class);
+    checkMustCallAliasAnnoMismatch(
+        paramWithMustCallAliasAnno, isMustCallAliasAnnoOnReturnType, tree);
+  }
+
+  /**
+   * Given a constructor, a {@code @MustCallAlias} must appear in both the list of parameters and as
+   * an annotation on the constructor itself, if it is to appear at all.
+   *
+   * <p>That is, a {@code @MustCallAlias} annotation must appear on both the constructor and its
+   * parameter list, or not at all.
+   *
+   * @param tree the constructor
+   */
+  private void checkMustCallAliasAnnotationForConstructor(MethodTree tree) {
+    ExecutableElement constructorDecl = TreeUtils.elementFromDeclaration(tree);
+    boolean isMustCallAliasAnnoOnConstructor =
+        constructorDecl != null && rlTypeFactory.hasMustCallAlias(constructorDecl);
+    Element paramWithMustCallAliasAnno = getParameterWithMustCallAliasAnno(tree);
+    checkMustCallAliasAnnoMismatch(
+        paramWithMustCallAliasAnno, isMustCallAliasAnnoOnConstructor, tree);
+  }
+
+  /**
+   * Construct the warning message for the case where a {@code @MustCallAlias} annotation does not
+   * appear in pairs in a method or constructor declaration.
+   *
+   * <p>If a parameter of a method or a constructor is annotated with a {@code @MustCallAlias}
+   * annotation, the return type (for a method) should also be annotated with
+   * {@code @MustCallAlias}. In the case of a constructor, which has no return type, a
+   * {@code @MustCallAlias} annotation must appear on its declaration.
+   *
+   * @param paramWithMustCallAliasAnno a parameter with a {@code @MustCallAlias} annotation, null if
+   *     there are none
+   * @param isMustCallAliasAnnoOnMethodOrConstructorDecl true if and only if a
+   *     {@code @MustCallAlias} annotation appears on a method or constructor declaration
+   * @param tree the method or constructor declaration
+   */
+  private void checkMustCallAliasAnnoMismatch(
+      @Nullable Element paramWithMustCallAliasAnno,
+      boolean isMustCallAliasAnnoOnMethodOrConstructorDecl,
+      MethodTree tree) {
+    boolean isMustCallAliasAnnotationOnParameter = paramWithMustCallAliasAnno != null;
+    String locationOfCheck = TreeUtils.isClassTree(tree) ? "this constructor" : "the return type";
```

Comment:
Move this computation inside the `if` statement, since it is not used until there.